### ISSUE TITLE
Support insecure skip verify for Tempo

### ISSUE THREAD

**Issue by bombsimon** [grafana/agent#257]
Currently both Prometheus and Loki configuration supports configuring insecure skip verify with the following alternatives.

```yaml
loki:
  clients:
    tls_config:
      insecure_skip_verify: true
prometheus:
  configs:
    remote_write:
      - tls_config:
        insecure_skip_verify: true
        scrape_configs:
          - job_name: some_name
            ...
            tls_config:
              insecure_skip_verify: true
```

The new configuration that should be added would be:

```yaml
tempo:
  push_config:
    insecure_skip_verify: true
```

This is currently not supported upstream in the `opentelemetry-collector` project but I've added [a PR](https://github.com/open-telemetry/opentelemetry-collector/pull/2202) to add support for this.

Sadly, this project uses an old tag of `opentelemetry-collector` (v0.14.0) and simply bumping the version won't work since v0.15.0 contains breaking changes.

I've prepared [a local branch](https://github.com/grafana/agent/compare/master...bombsimon:tls-insecure-skip-verify-tempo) to pick up if/when my PR is merged into `opentelemetry-collector`.

Are there any plans to bump `opentelemetry-collector` to v0.15.0? If not I can start to look into that as well since it will be required to get this new feature.

**Comment by rfratto** [comment]
Thanks for opening that upstream PR! We will definitely upgrade to the next otel collector version that includes your PR. 0.15.0 doesn’t seem to have anything in particular that the agent needs so we’ll likely skip that one, but that breaking change shouldn’t be a problem for when we do upgrade. 

Your Agent branch looks good and would be a quick LGTM from my side once your other PR is merged.
